packages/nvidia_nat_core/src/nat/utils/exception_handlers/automatic_retries.py

```
@@ -217,53 +238,39 @@ def decorate(fn: Callable[..., T]) -> Callable[..., T]:
         skip_self_in_deepcopy = instance_context_aware
 
         class _RetryContext:
-            """Context manager for instance-level retry gating."""
+            """Context manager for retry gating.
+
+            Scoped to the (object, owning task/thread) pair via the module-level
+            `_in_retry_context`, not to the patched instance alone."""
 
-            __slots__ = ("_obj_ref", "_enabled", "_active")
+            __slots__ = ("_key", "_enabled", "_token")
```

Ruff (0.16.2)

[warning] `_RetryContext.__slots__` is not sorted

Apply a natural sort to `_RetryContext.__slots__`

(RUF023)